WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsSource.kt

```diff
@@ -78,9 +81,14 @@ class CardsSource @Inject constructor(
         selectedSite: SiteModel
     ) {
         coroutineScope.launch(bgDispatcher) {
+            delay(REFRESH_DELAY)
             val result = cardsStore.fetchCards(selectedSite)
-            result.error?.let {
-                postState(CardsUpdate(result))
+            val model = result.model
+            val error = result.error
+            when {
+                error != null -> postState(CardsUpdate(result))
+                model != null -> refresh.postValue(false)
```

Comment:
Suggestion:How about adding a new method (or adjusting) the onRefreshed() method in the interface, so that refresh is never directly updated? This was the reason that onRefreshed was added originally. Wdyt?

Reply:
Nice idea! `onRefreshed()` adjusted in 9c028b5 and used only interface methods to update `refresh` value.